tools/Microsoft.AspNet.Server.Kestrel.GeneratedCode/FrameFeatureCollection.cs

```diff
@@ -25,26 +25,40 @@ public virtual void BeforeCompile(BeforeCompileContext context)
 
         public static string GeneratedFile()
         {
-            var commonFeatures = new[]
+            var alwaysFeatures = new[]
             {
                 typeof(IHttpRequestFeature),
                 typeof(IHttpResponseFeature),
                 typeof(IHttpRequestIdentifierFeature),
-                typeof(IHttpSendFileFeature),
                 typeof(IServiceProvidersFeature),
-                typeof(IHttpAuthenticationFeature),
                 typeof(IHttpRequestLifetimeFeature),
+                typeof(IHttpConnectionFeature)
+            };
+
+            var commonFeatures = new[]
+            {
+                typeof(IHttpAuthenticationFeature),
                 typeof(IQueryFeature),
-                typeof(IFormFeature),
+                typeof(IFormFeature)
+            };
+
+            var sometimesFeatures = new[]
+            {
+                typeof(IHttpUpgradeFeature),
```

Comment:
I called out these less common and rare features because I don't think they're worth first class treatment in this collection. The collection must handle arbitrary 3rd party features so these should just be lumped in with those.